automation/pg_si_test/pg_si_test.go

```diff
@@ -0,0 +1,280 @@
+	"github.com/nutanix-cloud-native/ndb-operator/automation"
+	clientsetv1alpha1 "github.com/nutanix-cloud-native/ndb-operator/automation/clientset/v1alpha1"
+	"github.com/nutanix-cloud-native/ndb-operator/common"
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_api"
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// A test suite is a collection of related test cases that are grouped together for testing a specific package or functionality.
+// The testify package builds on top of Go's built-in testing package and enhances it with additional features like assertions and test suite management.
+// PostgresqlSingleInstanceTestSuite is a test suite struct that embeds testify's suite.Suite
+type PostgresqlSingleInstanceTestSuite struct {
+	suite.Suite
+	config            *rest.Config
+	v1alpha1ClientSet *clientsetv1alpha1.V1alpha1Client
+	clientset         *kubernetes.Clientset
+	kubeconfig        string
+	logFile           string
+	setupPath         automation.SetupPaths
+}
+
+// SetupSuite is called once before running the tests in the suite
+func (suite *PostgresqlSingleInstanceTestSuite) SetupSuite() {
+	var err error
+	var config *rest.Config
+	var v1alpha1ClientSet *clientsetv1alpha1.V1alpha1Client
+	var clientset *kubernetes.Clientset
+	kubeconfig := os.Getenv("KUBECONFIG")
+	logFile := "./PostgresqlSingleInstanceTestSuite.log"
+
+	// Setup output log file
+	if _, err := os.Stat(logFile); err == nil {
+		_ = os.Remove(logFile)
+	}
+	file, err := os.OpenFile(logFile, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.SetOutput(file)
+
+	log.Printf("******************** RUNNING PostgresqlSingleInstanceTestSuite SETUPSUITE() ********************\n")
+
+	// Setup kubeconfig
+	if kubeconfig == "" {
+		log.Println("Using in-cluster configuration")
+		config, err = rest.InClusterConfig()
+	} else {
+		log.Printf("Using configuration from '%s'\n", kubeconfig)
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	}
+	if err != nil {
+		log.Printf("Error: %s\n", err)
+		suite.T().FailNow()
+	}
+
+	// Setup scheme and clientsets
+	ndbv1alpha1.AddToScheme(scheme.Scheme)
+	v1alpha1ClientSet, err = clientsetv1alpha1.NewForConfig(config)
+	if err != nil {
+		log.Printf("Error: %s\n", err)
+		suite.T().FailNow()
+	}
+	clientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Printf("Error: %s\n", err)
+		suite.T().FailNow()
+	}
+
+	// Create base setup for all tests in this suite
+	setupPaths := automation.SetupPaths{
+		DbSecretPath:  automation.DbSecretPath,
+		NdbSecretPath: automation.NdbSecretPath,
+		DbPath:        automation.DbPath,
```

Comment:
The paths will change here as well, since we will use different paths for different engines.